docs/source/advanced/advanced_gpu.rst

```diff
@@ -516,6 +525,26 @@ Here is some helpful information when setting up DeepSpeed ZeRO Stage 3 with Lig
 * When offloading to the CPU, make sure to bump up the batch size as GPU memory will be freed
 * We also support sharded checkpointing. By passing ``save_full_weights=False`` to the ``DeepSpeedPlugin``, we'll save shards of the model which allows you to save extremely large models. However to load the model and run test/validation/predict you must use the Trainer object.
 
+.. _deepspeed-zero-stage-3-single-file:
+
+Collating Single File Checkpoint for DeepSpeed ZeRO Stage 3
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+After training using ZeRO Stage 3, you'll notice that your checkpoints are a directory of sharded model and optimizer states. If you'd like to collate a single file from the checkpoint directory please use `this custom gist <https://gist.github.com/SeanNaren/e0535977df8e165247f7ff9ae445be1b>`__ which handles all the Lightning states additionally when collating the file.
+
+.. code-block:: bash
+
+    # enter directory where you saved your checkpoints
+    cd lightning_logs/version_0/checkpoints/
+    # lightning has saved a directory instead of a file
+    cd epoch=0-step=0.ckpt/
+
+    python convert_lightning_deepspeed_checkpoint.py . pytorch_model.ckpt
```

Comment:
Should we introduce this script functionality in Lightning?

Reply:
Great idea, lemme make a separate PR!